src/cs-core/Settings.cpp

```diff
@@ -329,6 +340,21 @@ void Settings::write(std::string const& fileName) const {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
+std::string Settings::saveToJson() const {
+  // Tell listeners that the settings are about to be saved.
+  mOnSave.emit();
+
+  nlohmann::json settings = *this;
+
+  // Use an indentation of two space.
+  std::ostringstream o;
+  o << std::setw(2) << settings;
```

Comment:
Why do we need the `setw` here? If the string contains more than two characters it shouldn't do anything. Or am I missing something?

Reply:
It's overloaded for nlohmann::json and specifies the indentation width.